pkg/process/checks/system_info_darwin_test.go

```diff
@@ -36,3 +37,19 @@ func TestPatchCPUInfo(t *testing.T) {
 		assert.Equal(t, int32(2), c.Cores)
 	}
 }
+
+func TestOnlyCorePopulatedDarwin(t *testing.T) {
+	sysInfo, _ := CollectSystemInfo()
+	for _, cpuData := range sysInfo.Cpus {
+		// Checks if only the cores does not have the default value
+		assert.NotEqual(t, int32(0), cpuData.Cores)
+		assert.Equal(t, int32(0), cpuData.Number)
+		assert.Equal(t, "", cpuData.Vendor)
+		assert.Equal(t, "", cpuData.Family)
+		assert.Equal(t, "", cpuData.Model)
+		assert.Equal(t, "", cpuData.PhysicalId)
+		assert.Equal(t, "", cpuData.CoreId)
+		assert.Equal(t, int64(0), cpuData.Mhz)
+		assert.Equal(t, int32(0), cpuData.CacheSize)
```

Comment:
You can use [assert.Empty()](https://pkg.go.dev/github.com/stretchr/testify/assert#Empty.

> Empty asserts that the specified object is empty. I.e. nil, "", false, 0 or either a slice or a channel with len == 0.